polynote-frontend/polynote/ui/component/cell.ts

```diff
@@ -896,7 +896,7 @@ export class CodeCell extends Cell {
                 this.container.classList.add('queued');
                 break;
             case "running":
-                this.container.classList.remove('queued', 'error');
+                this.container.classList.remove('queued');
```

Comment:
Can the cell be in an error state when it's running?

Reply:
I think the cell has to go through the queued state in order to transition to running, which will remove the error state from the cell.

Reply:
That status update might not necessarily be sent though (if it transitions through `Queued` too quickly). Is there any reason not to remove `error` here? It can't be in `error` if it's in `running`.

Reply:
Unfortunately we can get multiple task updates with the running state - and after the error result is added to the cell. So we'd get something like:
```
Queued
Running (without detail message in the task)
Add error result (not a state transition)
Running (add detail message in the task, maybe progress update)
Completed
```
The second Running update clears out the error result. 

The code isn't handling that case right now because it's not a proper state machine or anything. 

I didn't realize the status update might not be sent - I thought that if the transition happened quickly we'd just get the queued and running messages right after the other. 

If we can't rely on these status updates, I can look into making the UI logic robust to getting multiple running statuses. However I think we may want to revisit the Task State+Cell State+Error logic in the future as it's getting a bit messy.

Reply:
Agreed, we should add some kind of message (that isn't a status update) that a cell has finished (and whether or not it's errored). This is kind of tied up with the "protocol unification" effort, as the remote kernel protocol already has such a message (or at least a "this cell is done" message).

Anyway, as long as the cell's error state is cleared when you run it, I think we should be OK here for now. Is that the case?

Reply:
(by "run it" I mean click one of the run buttons in such a way that the server will be asked to queue the cell)

Reply:
Yep, the error state gets cleared up next time you run in my testing. But I just did super basic testing -  throw an error in a cell and check that running it again clears the error while it's running. Are there other scenarios you can think of which might transition differently?